Xamarin.Forms.Core/Shell/ShellSectionCollection.cs

```diff
@@ -119,7 +123,13 @@ void CheckVisibility(ShellSection section)
 
 		public void Add(ShellSection item) => Inner.Add(item);
 
-		public void Clear() => Inner.Clear();
+		public void Clear()
```

Comment:
This logic is duplicated in `ShellItemCollection.cs`,  maybe we can create a base class with this code (and remove logic)
What do you think? 😊

Reply:
It's probably about that time :-)
But I'll leave that for the next PR for now

There are type specifics that will probably need a few generics and probably some generic interfaces as well.

All good stuff but something better left for a later branch for now

Reply:
Maybe I can help with this, I've created a suggestion for this change.
What do you think?

https://github.com/xamarin/Xamarin.Forms/pull/9626